tutor/src/models/user.js

```diff
@@ -46,6 +46,18 @@ export class User extends BaseModel {
     return !!find(Courses.nonPreview.active.array, { canAnnotate: true });
   }
 
+  @computed get hasPreviewed() {
+    return Courses.preview.size == 0 || Courses.preview.isViewed.size > 0;
+  }
+
+  @computed get shouldPreview() {
+    const exploreViewStats = this.viewed_tour_stats.find((stat) => stat.id === 'explore-a-preview');
+    if (exploreViewStats) {
+      return exploreViewStats.view_count < 4;
+    }
+    return true;
+  }
+
   @lazyGetter annotations = new Annotations();
```

Comment:
This is fine, but I think if we add anymore preview methods we should add a "Previews" class and attach it to User with lazyGetter like we did for Annotations.